imap_processing/swe/l1a/swe_science.py

```diff
@@ -159,13 +159,29 @@ def swe_science(decom_data):
         attrs=cdf_attrs.get_variable_attributes("spin_angle"),
     )
 
+    # NOTE: LABL_PTR_1 should be CDF_CHAR.
+    spin_angle_label = xr.DataArray(
+        [str(i) for i in np.arange(180)],
```

Comment:
I would recommend generating this based on the values in `spin_angle` so that if those change, the label also changes. (This comment goes for all of the label variables)
I would think that CDAWeb should do this based on the DEPEND_i attributes. I will try to remember to ask Andriy about this next time.

Reply:
I think it's a good idea. I will update that in future PR if you don't mind.

Reply:
I think we should just do it here for explicitness and don't think it will be too hard to incorporate in this PR.

```suggestion
        spin_angle.astype(str),
```

Another question that this brings up is you have 180 bins here, but the angle resolution in the YAML file says 12 degrees, so I'm a bit confused about that mismatch.

Reply:
That `astype(str)` is a great idea. If that, then it may be easier change that what I was thinking initially.

180 is 180 measurement per quarter cycle. This is just for l1a. At l1a, it has not information about angle. What you are thinking is l1b data product description. It's binned in 30 bins and it's 12 degree nominally.

Reply:
This is the yaml descriptor you're referencing here. So I think you'd need to update this YAML description to be different for l1a and l1b then.
```
spin_angle_label:
  CATDESC: Spin angle step. Angle resolution is 12 degree nominally
  FIELDNAM: 12 Degree Resolution
  FORMAT: A3
  VAR_TYPE: metadata
```

Reply:
Ah ok. I will double check this. Thank you for catching it!